packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -203,6 +203,68 @@ class DateTimePicker extends React.Component {
     }
   }
 
+  getMetadata(momentDateTime) {
+    const isCompleteDateTime = DateTimeUtils.isValidDateTime(this.dateValue, this.timeValue, this.state.dateFormat, this.props.showSeconds);
+    let value = '';
+    let date = '';
```

Comment:
Now that this big blob of logic is extracted into a separate function, it could be shortened to get rid of a number of if checks.

```const dateTimeValue = `${dateValue ? dateValue : '' } ${timeValue ? timeValue : '' }`.trim();```

Since `date` and `time` are not being anywhere else but at the end to create `options` object, I'd do - 

```
const options = {
      ...
      dateValue: this.dateValue || '',
      timeValue: this.timeValue || '',
      ...
}
```

Reply:
Yes, this suggestion does help shorten and simplify the logic.
Addressed it here : https://github.com/cerner/terra-framework/pull/1045/commits/1e8dc74b082dc0de2c1002be51b425e1d49f9784